cmd/vulcan-gitleaks/main.go

```diff
@@ -0,0 +1,237 @@
+			return err
+		}
+	}
+
+	if opt.Depth == 0 {
+		opt.Depth = DefaultDepth
+	}
+
+	logger.WithFields(logrus.Fields{"options": opt}).Debug("using options")
+
+	// We check if the target is not the public Github.
+	targetURL, err := url.Parse(target)
+	if err != nil {
+		return err
+	}
+
+	// TODO: Support multiple authenticated Github Enterprise instances.
+	githubURL, err := url.Parse(os.Getenv("GITHUB_ENTERPRISE_ENDPOINT"))
+	if err != nil {
+		return err
+	}
+
+	var auth *http.BasicAuth
+	if githubURL.Host != "" && targetURL.Host == githubURL.Host {
+		auth = &http.BasicAuth{
+			Username: "username", // Can be anything except blank.
+			Password: os.Getenv("GITHUB_ENTERPRISE_TOKEN"),
+		}
+		logger.Debug("using credentials for GitHub")
+	}
+
+	gitCreds := &helpers.GitCreds{}
+	if auth != nil {
+		gitCreds.User = auth.Username
+		gitCreds.Pass = auth.Password
+	}
+	isReachable, err := helpers.IsReachable(target, assetType, gitCreds)
+	if err != nil {
+		logger.WithError(err).Warn("can not check asset reachability")
+	}
+	if !isReachable {
+		return checkstate.ErrAssetUnreachable
+	}
+
+	repoPath := filepath.Join(os.TempDir(), "repo")
+	if err := os.Mkdir(repoPath, 0755); err != nil {
+		return err
+	}
+
+	logger.WithFields(logrus.Fields{"repo_path": repoPath}).Debug("cloning repo")
+
+	co := git.CloneOptions{
+		URL:   target,
+		Auth:  auth,
+		Depth: opt.Depth,
+	}
+	if opt.Branch != "" {
+		co.ReferenceName = plumbing.ReferenceName(path.Join("refs/heads", opt.Branch))
+	}
+	r, err := git.PlainClone(repoPath, false, &co)
+	if err != nil {
+		return err
+	}
+
+	b, err := r.Head()
+	if err != nil {
+		logger.Errorf("vulcan-gitleaks was not able to retrieve the working branch")
+		return err
+	}
+
+	// Run gitleaks.
+	err = runGitleaks(ctx, logger, repoPath)
+	if err != nil {
+		return err
+	}
+
+	// Read the results file
+	byteValue, err := ioutil.ReadFile(reportOutputFile)
+	if err != nil {
+		logger.Errorf("gitleaks report output file read failed with error: %s\n", err)
+		return errors.New("gitleaks report output file read failed")
+	}
+
+	var results []Finding
+	err = json.Unmarshal(byteValue, &results)
+	if err != nil {
+		return errors.New("unmarshal gitleaks output failed")
+	}
+
+	// Process the secrets found by gitleaks.
+	return processVulns(results, opt, repoPath, string(b.Name()), target, state)
+}
+
+func processVulns(results []Finding, opt options, repoPath string, branch string, target string, state checkstate.State) error {
+	// Return if there are no findings.
+	if len(results) < 1 {
+		return nil
+	}
+
+	// Group secrets by file.
+	for _, f := range results {
+		if stringInSlice(f.RuleID, &opt.ExcludedRules) {
+			continue
+		}
+		file := strings.TrimPrefix(f.File, repoPath)
+		v := leakedSecret
+		s, _ := bcrypt.GenerateFromPassword([]byte(f.Secret), 0)
+		v.AffectedResource = string(s)
+		v.AffectedResourceString = computeAffectedResource(target, branch, file, f.StartLine)
+		v.Fingerprint = helpers.ComputeFingerprint("")
```

Comment:
Not relevant at all, but I'd suggest removing the `""` and just call the method without args, to be consistent with what has been implemented in the other checks.

Reply:
Completelly agree, fixed here [ec07e44](https://github.com/adevinta/vulcan-checks/pull/185/commits/ec07e443a7a05c55ec762fc6ffe12642f82d8256).